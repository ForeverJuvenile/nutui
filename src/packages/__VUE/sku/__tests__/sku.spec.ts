import { mount } from '@vue/test-utils';
import { nextTick } from 'vue';
import Sku from '../index.vue';
import { Sku as SkuData, Goods } from '../data';

test('sku init render', async () => {
  const wrapper = mount(Sku, {
    props: {
      visible: true,
      teleportDisable: false,
      sku: SkuData,
      goods: Goods
    }
  });
  await nextTick();
  expect(wrapper.find('.nut-sku-select').exists()).toBeTruthy();
});

test('sku select event', async () => {
  const wrapper = mount(Sku, {
    props: {
      visible: true,
      teleportDisable: false,
      sku: SkuData,
      goods: Goods
    }
  });
  await nextTick();
  const skuItem = wrapper.findAll('.nut-sku-select-item-skus-sku');
  skuItem[1].trigger('click');
  const selectSkuObj = wrapper.emitted().selectSku[0] as [];
  expect(selectSkuObj.length).toBe(1);
});

test('do not sell', async () => {
  const wrapper = mount(Sku, {
    props: {
      visible: true,
      teleportDisable: false,
      sku: SkuData,
      goods: Goods,
      btnExtraText: '抱歉，此商品在所选区域暂无存货'
    }
  });
  await nextTick();
  expect(wrapper.find('.nut-sku-operate-desc').exists()).toBeTruthy();
  expect(wrapper.find('.nut-sku-operate-desc').text()).toEqual('抱歉，此商品在所选区域暂无存货');
});

test('button event', async () => {
  const wrapper = mount(Sku, {
    props: {
      visible: true,
      teleportDisable: false,
      sku: SkuData,
      goods: Goods
    }
  });
  await nextTick();
  expect(wrapper.find('.nut-sku-operate-btn-confirm').exists()).toBeTruthy();
  wrapper.find('.nut-sku-operate-btn-confirm').trigger('click');
  expect(wrapper.emitted().clickBtnOperate[0]).toEqual([{ type: 'confirm', value: 1 }]);
});
