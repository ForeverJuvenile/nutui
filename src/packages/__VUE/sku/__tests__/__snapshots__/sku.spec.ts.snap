// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sku select event 1`] = `
[
  {
    "parentIndex": 0,
    "parentSku": {
      "id": 1,
      "list": [
        {
          "active": true,
          "disable": false,
          "id": 100016015112,
          "name": "亮黑色",
        },
        {
          "active": false,
          "disable": false,
          "id": 100016015142,
          "name": "釉白色",
        },
        {
          "active": false,
          "disable": false,
          "id": 100016015078,
          "name": "秘银色",
        },
        {
          "active": false,
          "disable": false,
          "id": 100009064831,
          "name": "夏日胡杨",
        },
        {
          "active": false,
          "disable": false,
          "id": 100009064830,
          "name": "秋日胡杨",
        },
      ],
      "name": "颜色",
    },
    "sku": {
      "active": false,
      "disable": false,
      "id": 100016015142,
      "name": "釉白色",
    },
    "skuIndex": 1,
  },
]
`;
